### PR TITLE
fix(k8s): make sure controllers start after leader election

### LINKED ISSUE #3611: Leader election seems broken for Kubernetes
### What happened?

When updating `kuma-cp`, the new control plane starts the `controller-runtime.Manager` immediately, leading to reconciliation errors because of concurrent resource updates.

This likely has to do with `LeaderElection` being disabled in the `controller-runtime.Manager`. It may be possible to hook our leader-for-life election into the `Manager`.

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Summary

This PR adds an intercept to `Manager.Add` calls to save `Controller`s to be `Add`ed after leader election.

Closes #3611

### CODE REVIEW

File: pkg/plugins/bootstrap/k8s/plugin.go [hunk 1]
```diff
@@ -179,6 +220,18 @@ func makeOldLockAnnotation() string {
 	return string(annotJson)
 }
 
+// startLeaderComponents adds all components that need leader election to the
+// Manager. The Manager should be running already, any components added after
+// that are started immediately.
+func (cm *kubeComponentManager) startLeaderComponents() {
```
Comment (jakubdyszkiewicz) [inline pkg/plugins/bootstrap/k8s/plugin.go:226]: I think any error in this function should be returned, logged and then `os.Exit(1)`.

Comment (michaelbeaumont) [reply pkg/plugins/bootstrap/k8s/plugin.go:226]: Good point, maybe we should return an `error` here and then return that error in `Start(done <-chan struct{}) error` further down? That should trigger `kuma-cp` to exit.

Comment (jakubdyszkiewicz) [reply pkg/plugins/bootstrap/k8s/plugin.go:226]: Yes, if it's possible to get rid of `os.Exit(1)`, that's even better.

Comment (michaelbeaumont) [reply pkg/plugins/bootstrap/k8s/plugin.go]: Done. Used `golang.org/x/sync`. It's also possible with just channels, `errgroup` is nicer to read I think.

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**